src/Client.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Baka\Elasticsearch;
+
+use Exception;
+use GuzzleHttp\Client as GuzzleClient;
+
+class Client
+{
+    private $host;
+
+    /**
+     * Set the host.
+     *
+     * @param string $host
+     * @return void
+     */
+    public function __construct(string $host)
+    {
+        $this->host = $host;
+    }
+
+    /**
+     * Given a SQL search the elastic indices.
+     *
+     * @param string $sql
+     * @return void
+     */
+    public function findBySql(string $sql): array
+    {
+        $client = new GuzzleClient([
+            'base_uri' => $this->host,
+        ]);
+
+        // since 6.x we need to use POST
+        $response = $client->post('/_sql', [
```

Comment:
@kaioken Put this inside of try-catch, if the URL can't be reach or the connection could be established, an exception will be thrown.

Reply:
By default guzzle throws the exception I was planning on capturing it on the API controller

Reply:
Yes, `an exception will be thrown`.
I think it should be handled in the package itself and throw a custom elasticsearch-package-exception. In that way, whoever uses the package will get an Elastic related exception and will determinate how to handle it. 
For example, if the S3 SDK throw an exception and I'm using flysystem, the exception that I get in my app is from flysystem not from S3 SDK.
I think you can have more control and context if the exception is from the package you are using and not the packages the library use.

Reply:
Will see on how to handle it than wen I get to the complex canvas filter